Aptoide/aptoide-client-v8
# Feature/asv 1047 api for videos

**What does this PR do?**

This PR was supposed to implement calls to the videos API. However, it would be impossible to properly test those calls without implementing the rest of the logic (endless scrolling, showing button in home, correctly parsing videos, etc...) 

**Database changed?**

No

**Where should the reviewer start?**

- [ ] RemoteVideoDataSource.java
- [ ] VideosPresenter.kt

**How should this be manually tested?**

Check the videos fragment. To do this you need to mock responses for the two webservices: the one asking if there's videos and the one that actually returns the video list. To mock the first one:

Endpoint:
`ws75.aptoide.com/api/7/community/videos/count`
Response:
`{"info":{"status":"OK","time":{"seconds":0.059432029724121,"human":"59 milliseconds"}},"data":{"live_streaming":1,"non_streaming":1},"ab_testing":{"name":"<str>","assignment":"<str>"}}`

The second one:

Endpoint:
`ws75.aptoide.com/api/7/community/videos/get`
Response:
`{"info":{"status":"OK","time":{"seconds":0.059432029724121,"human":"59 milliseconds"}},"datalist":{"total":90,"count":10,"offset":0,"limit":10,"next":10,"hidden":0,"loaded":true,"list":[{"type":"","url":"http://videocdn.bodybuilding.com/video/mp4/62000/62792m.mp4","thumbnail":"<str>","app":{"stats":{"rating":{"avg":4,"total":4628}},"id":38590960,"name":"Game #3","package":"com.milliongames.sanandreasautocrimetheft","uname":"san-andreas-auto-crime-theft","size":91274387,"icon":"http://java.sogeti.nl/JavaBlog/wp-content/uploads/2009/04/android_icon_256.png","graphic":"null"},"date":"2018-09-22 17:52:23","source":"<str>","current_viewers":10,"total_viewers":10},{"type":"","url":"rtsp://wowzaec2demo.streamlock.net/vod/mp4:BigBuckBunny_115k.mov","thumbnail":"<str>","app":{"stats":{"rating":{"avg":3.9,"total":4628}},"id":38590960,"name":"Game #2","package":"com.milliongames.sanandreasautocrimetheft","uname":"san-andreas-auto-crime-theft","size":91274387,"icon":"https://lh3.googleusercontent.com/rFmTpCoFfUraFMJA1oVj3IzgHMcadJYn3C_H2FTTrgHVBRVoyqXPsyX6uCzRmduxtaI=s180-rw","graphic":"null"},"date":"2018-09-22 17:52:23","source":"<str>","current_viewers":10,"total_viewers":10},{"type":"","url":"http://videocdn.bodybuilding.com/video/mp4/62000/62792m.mp4","thumbnail":"<str>","app":{"stats":{"rating":{"avg":4.21,"total":4628}},"id":38590960,"name":"Game #1","package":"com.milliongames.sanandreasautocrimetheft","uname":"san-andreas-auto-crime-theft","size":91274387,"icon":"https://lh3.googleusercontent.com/G0Kaa6WRog41pCTPQKlPIgLOiEl-RpmVDrUQySGMT3o9DwOl4SSLENgEVTQIRQv9QQF-=s180-rw","graphic":"null"},"date":"2018-09-22 17:52:23","source":"<str>","current_viewers":10,"total_viewers":10}]}}`

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1047](<https://aptoide.atlassian.net/browse/ASV-1047>)

**Code Review Checklist**
